eureka-client/src/main/java/com/netflix/discovery/converters/EurekaJacksonCodec.java

```diff
@@ -94,22 +103,30 @@
     /**
      * XStream codec supports character replacement in field names to generate XML friendly
      * names. This feature is also configurable, and replacement strings can be provided by a user.
-     * To obey these rules, version and apppsHash key field names must be formatted according to the provided
+     * To obey these rules, version and appsHash key field names must be formatted according to the provided
      * configuration, which by default replaces '_' with '__' (double underscores).
      */
     private final String versionDeltaKey;
     private final String appHashCodeKey;
 
     private final ObjectMapper mapper;
 
-    private final Map<Class<?>, ObjectReader> objectReaderByClass;
+    private final Map<Class<?>, Supplier<ObjectReader>> objectReaderByClass;
     private final Map<Class<?>, ObjectWriter> objectWriterByClass;
 
+    static EurekaClientConfig loadConfig() {
+        return com.netflix.discovery.DiscoveryManager.getInstance().getEurekaClientConfig();
```

Comment:
that's existing code, relocated to an out-of-the-way method that preserves backward compatibility.  For DI, use the two-arg constructor that requires neither DiscoveryManager nor EurekaClientConfig.